docs/source/api/apollo-gateway.mdx

```diff
@@ -416,3 +416,86 @@ This enables you to modify any combination of the operation's `context` and the
 The `http` property of the `request` and `response` objects contains additional HTTP-specific properties, such as `headers`.
 
 This method must return an object that matches the structure of a [`GraphQLResponse`](https://github.com/apollographql/apollo-server/blob/2562d096/packages/apollo-server-types/src/index.ts#L43-L48).  If no modifications are necessary, return the original `response`.
+
+#### `didEncounterError`
+
+Override this method in a subclass to customize the gateway's behavior after it encounters an error while communicating with the implementing service.
+
+If you _don't_ override this method, the default implementation `throw`s the error, like so:
+
+```javascript
+class MyDataSource extends RemoteGraphQLDataSource {
+
+  didEncounterError({ error, fetchRequest, fetchResponse, context }) {
+    throw error;
+  }
+
+}
+```
+
+Note that if you _don't_ `throw error` (or a different `Error`) in this method, `error` is thrown immediately after this method returns.
+
+This method takes the following parameters:
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+<tr>
+<td>
+
+###### `error`
+
+`Error`
+</td>
+<td>
+
+The error that occurred during communication.
+</td>
+</tr>
+
+<tr>
+<td>
+
+###### `fetchRequest`
+
+`Request`
+</td>
+<td>
+
+The details of the `fetch` request sent to the implementing service.
+</td>
+</tr>
+
+<tr>
+<td>
+
+###### `fetchResponse`
+
+`Response`
+</td>
+<td>
+
+The details of the `fetch` response sent by the implementing service.
+</td>
+</tr>
+
+
+<tr>
+<td>
+
+###### `context`
+
+`TContext`
+</td>
+<td>
+
+The context of the communication's associated GraphQL request.
```

Comment:
Not suggesting that this needs to be annotated differently, but this is the "user-context" (if we have a notion of that anywhere).  Essentially, it's the thing that a user might have defined using the `context` function at the top-level of the `ApolloServer` constructor (where implementors might attach authentication details, etc., and would be available to resolvers in non-Gateway execution).